devtools/onion.c

```diff
@@ -58,49 +57,31 @@ static void do_generate(int argc, char **argv,
 			     argv[1 + i]);
 		}
 
-		memset(&hops_data[i], 0, sizeof(hops_data[i]));
-		if (argv[1 + i][klen] != '\0') {
-			/* FIXME: Generic realm support, not this hack! */
-			/* FIXME: Multi hop! */
+		/* /<hex> -> raw hopdata. /tlv -> TLV encoding. */
+		if (argv[1 + i][klen] != '\0' && argv[1 + i][klen] != 't') {
```

Comment:
This seems to undo some of the fixes from #3250 (https://github.com/ElementsProject/lightning/pull/3250/files#diff-6ba0afe5becbb580348286dbd8ca2b5cR63). Which version is the correct one?

Reply:
I've just rebased again on master, which fixes this...